server/controllers/Followers.js

```diff
@@ -0,0 +1,151 @@
+import models from '../database/models';
+import { serverResponse, serverError } from '../helpers';
+
+const { User, UserFollower } = models;
+
+/**
+ * @export
+ * @class Followers
+ */
+class Followers {
+  /**
+   * @name follow
+   * @async
+   * @static
+   * @memberof Users
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new follower
+   */
+  static async follow(req, res) {
+    try {
+      const {
+        user: { id },
+        params: { username }
+      } = req;
+      const user = await User.findByUsername(username);
+      if (!user) {
+        return serverResponse(res, 404, { error: 'user not found' });
+      }
+      if (id === user.id) {
+        return serverResponse(res, 409, {
+          error: 'user cannot perform this action'
+        });
+      }
+      const follower = await UserFollower.create({
+        userId: user.id,
+        followerId: id
+      });
+      const { dataValues } = follower;
+      serverResponse(res, 200, { user: dataValues });
+    } catch (error) {
+      return serverError(res);
+    }
+  }
+
+  /**
+   * @name unfollow
+   * @async
+   * @static
+   * @memberof Users
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of an unfollowed user
+   */
+  static async unfollow(req, res) {
+    try {
+      const {
+        user: { id },
+        params: { username }
+      } = req;
+      const user = await User.findByUsername(username);
+      if (!user) {
+        return serverResponse(res, 404, { error: 'user not found' });
+      }
+      if (id === user.id) {
+        return serverResponse(res, 409, {
+          error: 'user cannot perform this action'
+        });
+      }
+      await UserFollower.destroy({
+        where: {
+          userId: user.id,
+          followerId: id
+        }
+      });
+      serverResponse(res, 200, {
+        message: `you sucessfully unfollowed ${username}`
+      });
+    } catch (error) {
+      return serverError(res);
+    }
+  }
+
+  /**
+   * @name allFollowers
+   * @async
+   * @static
+   * @memberof Users
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of an unfollowed user
+   */
+  static async allFollowers(req, res) {
+    try {
+      const {
+        user: { id }
+      } = req;
+      const user = await User.findByPk(id);
+      const followers = await user.getAllFollowers({
+        include: [
+          {
+            model: User,
+            as: 'follower'
+          }
+        ]
+      });
+      const users = followers.map((x) => {
```

Comment:
never use single characters for parameter names unless it's a counter, change to `x` to `follower`

Reply:
okay. Noted.